tests/aws/services/cloudwatch/test_cloudwatch.py

```diff
@@ -1659,49 +1717,39 @@ def test_get_metric_data_with_zero_and_labels(self, aws_client):
-        response = aws_client.cloudwatch.get_metric_data(
-            MetricDataQueries=[
-                {
-                    "Id": "result_" + stat,
-                    "MetricStat": {
-                        "Metric": {"Namespace": namespace1, "MetricName": "metric1"},
-                        "Period": 60,
-                        "Stat": stat,
-                    },
-                }
-                for stat in stats
-            ],
-            StartTime=utc_now - timedelta(seconds=60),
-            EndTime=utc_now + timedelta(seconds=60),
-        )
-        #
-        # Assert Average/Min/Max/Sum is returned as expected
-        avg = [res for res in response["MetricDataResults"] if res["Id"] == "result_Average"][0]
-        assert avg["Label"] == "metric1 Average"
-        assert avg["StatusCode"] == "Complete"
-        assert [int(val) for val in avg["Values"]] == [19]
-
-        sum_ = [res for res in response["MetricDataResults"] if res["Id"] == "result_Sum"][0]
-        assert sum_["Label"] == "metric1 Sum"
-        assert sum_["StatusCode"] == "Complete"
-        assert [val for val in sum_["Values"]] == [sum(values)]
-
-        min_ = [res for res in response["MetricDataResults"] if res["Id"] == "result_Minimum"][0]
-        assert min_["Label"] == "metric1 Minimum"
-        assert min_["StatusCode"] == "Complete"
-        assert [int(val) for val in min_["Values"]] == [0]
-
-        max_ = [res for res in response["MetricDataResults"] if res["Id"] == "result_Maximum"][0]
-        assert max_["Label"] == "metric1 Maximum"
-        assert max_["StatusCode"] == "Complete"
-        assert [int(val) for val in max_["Values"]] == [100]
+
+        def _get_metric_data():
+            return aws_client.cloudwatch.get_metric_data(
+                MetricDataQueries=[
+                    {
+                        "Id": "result_" + stat,
+                        "MetricStat": {
+                            "Metric": {"Namespace": namespace1, "MetricName": "metric1"},
+                            "Period": 60,
+                            "Stat": stat,
+                        },
+                    }
+                    for stat in stats
+                ],
+                StartTime=utc_now - timedelta(seconds=60),
+                EndTime=utc_now + timedelta(seconds=60),
+            )
+
+        def _match_results():
+            response = _get_metric_data()
+            # keep one assert to avoid storing incorrect values
+            avg = [res for res in response["MetricDataResults"] if res["Id"] == "result_Average"][0]
+            assert [int(val) for val in avg["Values"]] == [19]
+            snapshot.match("get_metric_data_with_zero_and_labels", response)
+
+        retry(_match_results, retries=10, sleep=1.0)
 
     @markers.aws.validated
-    def test_get_metric_statistics(self, aws_client):
+    @markers.snapshot.skip_snapshot_verify(paths=["$..Datapoints..Unit"], condition=is_new_provider)
```

Comment:
my bad: this test is failing with the old provider as well, we need the skip for both providers.